internal/service/bedrockagent/agent.go

```diff
@@ -0,0 +1,528 @@
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource(name="Bedrock Agent")
+// @Tags(identifierAttribute="agent_arn")
+func newBedrockAgentResource(context.Context) (resource.ResourceWithConfigure, error) {
+	r := &bedrockAgentResource{}
+
+	r.SetDefaultCreateTimeout(5 * time.Minute)
+	r.SetDefaultUpdateTimeout(5 * time.Minute)
+	r.SetDefaultDeleteTimeout(5 * time.Minute)
+
+	return r, nil
+}
+
+type bedrockAgentResource struct {
+	framework.ResourceWithConfigure
+	framework.WithImportByID
+	framework.WithTimeouts
+}
+
+func (r *bedrockAgentResource) Metadata(_ context.Context, request resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_bedrockagent_agent"
+}
+
+func (r *bedrockAgentResource) Schema(ctx context.Context, request resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"agent_arn": framework.ARNAttributeComputedOnly(),
+			"agent_id":  framework.IDAttribute(),
+			"agent_name": schema.StringAttribute{
+				Required: true,
+			},
+			"agent_resource_role_arn": schema.StringAttribute{
+				CustomType: fwtypes.ARNType,
+				Required:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"agent_status": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"agent_version": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"created_at": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"customer_encryption_key_arn": schema.StringAttribute{
+				CustomType: fwtypes.ARNType,
+				Optional:   true,
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"failure_reasons": schema.ListAttribute{
+				ElementType: types.StringType,
+				Computed:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"foundation_model": schema.StringAttribute{
+				Required: true,
+			},
+			names.AttrID: framework.IDAttribute(),
+			"idle_ttl": schema.Int64Attribute{
+				Required: true,
+			},
+			"instruction": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"prepared_at": schema.StringAttribute{
+				CustomType: timetypes.RFC3339Type{},
+				Computed:   true,
+			},
+			"prompt_override_configuration": schema.ListAttribute{ // Limited here by V5 Protocol
+				Computed:   true,
+				Optional:   true,
+				Validators: []validator.List{listvalidator.SizeAtMost(1)},
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+				CustomType: fwtypes.NewListNestedObjectTypeOf[poc](ctx),
+				ElementType: types.ObjectType{
```

Comment:
This follows the guidance for blocks with computed attributes for V5 Protocol, it could be nicer if the provider were using V6 protocol.

Reply:
💯 %